doc/contribute/guidelines.rst

```diff
@@ -573,12 +573,14 @@ workflow here:
    you want to open a pull request with.
 
 #. Review the pull request changes, and verify that you are opening a
-   pull request for the appropriate branch. The title and message from your
-   commit message should appear as well.
-
-#. If you're working on a subsystem branch that's not ``main``,
-   you may need to change the intended branch for the pull request
-   here, for example, by changing the base branch from ``main`` to ``net``.
+   pull request for the appropriate branch (if unsure, use `main`). The title
+   and message from your commit message should appear as well.
+
+   .. note:: As more commits are merged upstream, the GitHub PR page will show
+      a ``This branch is out-of-date with the base branch`` message and a
+      ``Update branch`` button on the PR page. That message should be ignored,
+      as triggering a branch update from the GitHub UI will cause the PR
+      approvals to be dropped, and a merge commit to be added to the PR.
```

Comment:
> triggering a branch update from the GitHub UI will cause [...] and a merge commit to be added to the PR.

That is not true if you choose "Update with rebase."

While I agree that the message should be ignored, I do not think we are providing the correct rationale for it here.

Reply:
> I could drop the merge commit bit and say "That message should be ignored, as the commits will be rebased as part of merging anyway, and triggering a branch update from the GitHub UI will cause the PR approvals to be dropped."

That sounds reasonable.

Reply:
Cool, reworded.